metadata/src/main/java/org/apache/kafka/controller/FeatureControlManager.java

```diff
@@ -370,6 +371,10 @@ public void replay(FeatureLevelRecord record) {
             MetadataVersion mv = MetadataVersion.fromFeatureLevel(record.featureLevel());
             metadataVersion.set(Optional.of(mv));
             log.info("Replayed a FeatureLevelRecord setting metadata.version to {}", mv);
+        } else if (record.name().equals(KRaftVersion.FEATURE_NAME)) {
+            // Skip any feature level record for kraft.version. This has two benefits:
+            // 1. It removes from snapshots any FeatureLevelRecord for kraft.version that was incorrectly written to the log
```

Comment:
can you add a reference to this JIRA in this comment?

Reply:
Fixed.